metrics/core/src/main/java/org/hyperledger/besu/metrics/opentelemetry/OpenTelemetrySystem.java

```diff
@@ -0,0 +1,307 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.metrics.opentelemetry;
+
+import org.hyperledger.besu.metrics.BesuMetricCategory;
+import org.hyperledger.besu.metrics.ObservableMetricsSystem;
+import org.hyperledger.besu.metrics.Observation;
+import org.hyperledger.besu.metrics.StandardMetricCategory;
+import org.hyperledger.besu.metrics.noop.NoOpMetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.Counter;
+import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
+import org.hyperledger.besu.plugin.services.metrics.MetricCategory;
+import org.hyperledger.besu.plugin.services.metrics.OperationTimer;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryUsage;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.DoubleSupplier;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableSet;
+import io.opentelemetry.common.Attributes;
+import io.opentelemetry.common.Labels;
+import io.opentelemetry.exporters.otlp.OtlpGrpcMetricExporter;
+import io.opentelemetry.metrics.DoubleValueObserver;
+import io.opentelemetry.metrics.DoubleValueRecorder;
+import io.opentelemetry.metrics.LongCounter;
+import io.opentelemetry.metrics.LongSumObserver;
+import io.opentelemetry.metrics.LongUpDownSumObserver;
+import io.opentelemetry.metrics.Meter;
+import io.opentelemetry.sdk.metrics.MeterSdkProvider;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.IntervalMetricReader;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.resources.ResourceAttributes;
+
+/** Metrics system relying on the native OpenTelemetry format. */
+public class OpenTelemetrySystem implements ObservableMetricsSystem {
+  private static final String TYPE_LABEL_KEY = "type";
+  private static final String AREA_LABEL_KEY = "area";
+  private static final String POOL_LABEL_KEY = "pool";
+  private static final String USED = "used";
+  private static final String COMMITTED = "committed";
+  private static final String MAX = "max";
+  private static final String HEAP = "heap";
+  private static final String NON_HEAP = "non_heap";
+
+  private final Set<MetricCategory> enabledCategories;
+  private final boolean timersEnabled;
+  private final Map<String, LabelledMetric<Counter>> cachedCounters = new ConcurrentHashMap<>();
+  private final Map<String, LabelledMetric<OperationTimer>> cachedTimers =
+      new ConcurrentHashMap<>();
+  private final MeterSdkProvider meterSdkProvider;
+  private IntervalMetricReader periodicReader;
```

Comment:
The handling of the network communication tasks for Prometheus is separated into the MetricsService, which handles lifecycle events as well as port management (if that's needed).

Rather than pull Promethous' networking stack into the PromethousMetricSystem we should pull at least the networking lifecycle and port management parts into a `MetricsService` class.  If pull is a thing that would include hooks to start pull as well as the existing push code in response to `start`, do what is done in `close` in the `stop` method, and if reporting a port makes sense (pull only) report it in getPort.  Ideally the networking classes would live in this new class (just periodicReader?) but I am OK with the OLTPMetricsService simply doing what needs to be done in this class via method calls.

To that end `MetricsService.create` will need some adjustment, have it return an `Optional<MetricsService>` and the existing code goes down the stack where protocol==PROMETHEOUS and the catch-all (for NONE) returns an `Optional.empty()`, also the calls that have guards can have those guards removed as `create` does it all now.

Reply:
OK, I think I have done that.